### ISSUE TITLE
new rule: enforce one of the two Underscore function call forms

### ISSUE THREAD

**Issue by danielmanesku** [Microsoft/tslint-microsoft-contrib#44]
What do you think of creating a rule for enforcing one of the two forms of Underscore functions? Should we create a rule for this?
```javascript
_.min(list, function() {...});
_(list).min(function() {...});
```

Personally, I prefer the second form.

**Comment by HamletDRC** [comment]
Sounds good to me. I prefer the 2nd one too.

**Comment by loicraux** [comment]
Does this apply to all Underscore methods ? Or should it apply to only a subset of the methods (like the ones for collections (http://underscorejs.org/#collections) ?

**Comment by danielmanesku** [comment]
I would say collections and arrays (at least)

**Comment by loicraux** [comment]
Not sure if we can enable this rule... :

I just discovered that this code does not build :
```typescript
function foo(dic: _.Dictionary<number>) : void {
    _(dic).forEach((val: number, key: string): void => {});
}
```
Error is :
```
error TS2345: Argument of type '(val: number, key: string) => void' is not assignable to
parameter of type 'ObjectIterator<Dictionary<number>, void>'.
  Types of parameters 'val' and 'element' are incompatible.
    Type 'number' is not assignable to type 'Dictionary<number>'.
      Index signature is missing in type 'Number'.
```
Whereas this one does build :
```typescript
function foo(dic: _.Dictionary<number>) : void {
    _.forEach(dic, (val: number, key: string): void => {});
}
```
I don't know if this is an error in underscore.d.ts ... I have to check.

**Comment by danielmanesku** [comment]
Might be a bug in our d.ts. file.

This works: 
```javascript
_.forEach([1, 2, 3], function(num) { console.log(num); })
_([5, 6, 7]).forEach(function(num) { console.log(num); })
```
Test it here: https://jsfiddle.net/kms3y8b8/. Enable developer tools to see the output and click on Run.
